app/controllers/comments_controller.rb

```diff
@@ -30,6 +21,14 @@ def create
     end
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    @post = Post.find(@comment.post_id)
+    @post.decrement!(:comment_counter)
+    @comment.destroy
+    redirect_to user_post_url(@post.author, @post), notice: 'Comment was successfully destroyed.'
+  end
```

Comment:
- [ ] [ OPTIONAL] I tried to comment and also delete but I can not perform both operations.  I would like to strongly recommend you make a user able to delete a post and or a comment if it is theirs or if they have an admin role to make your application works as expected, Please note that it is very important to follow the requirement :smile: